src/dev/preflight_check/utils/get_diffed_files_for_current_branch.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import simpleGit from 'simple-git';
+import { existsSync } from 'fs';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { nonNullable } from './non_nullable';
+
+const HASH_FOR_NULL_OR_DELETED_FILE = '00000000000';
+
+export async function getDiffedFilesForCurrentBranch() {
+  const git = simpleGit(REPO_ROOT);
+
+  const { current } = await git.branchLocal();
+
+  const commonAncestor = (await git.raw(['merge-base', 'origin/main', current])).trim();
```

Comment:
Thoughts on making `origin` configurable and/or able to use local branches?  

> ERROR ❌ 30510 committed files changed on this branch compared to origin/main and max files is set to 30.

My `un-updated` origin aside, I do somewhat regularly work off other developer's remotes.

Reply:
Existing discussion: https://github.com/elastic/kibana/pull/174831#discussion_r1452453704